features/step-definitions/task.2.js

```diff
@@ -0,0 +1,144 @@
+
+const { Given, When, Then } = require('@cucumber/cucumber');
+const HomePage = require('../pageobjects/home.page');
+const HoverPage = require('../pageobjects/hover.page');
+const AlertPage=require('../pageobjects/alert.page');
+const send_text='sadia'
+const image_number=2;
+
+//test case 3
+
+Given(/^I am on the Home page$/,async () => {

```

Comment:
You already have that step in another file. The purpose of using cucmber is to get rid of the repeating code.
If you have two steps with the same name the cucumber doesn't always knows what to choose and just dies. That's why you can see random failures with seemingly no reason to them. You had the same thing in task 1, I just didn't notice.

Reply:
Not resolved. This step is still in the task.1.js file, which results in instability and skipping some of the time